thrust/thrust/optional.h

```diff
@@ -2776,10 +2776,11 @@ class optional<T&>
   template <class... Args>
   _CCCL_HOST_DEVICE T& emplace(Args&&... args) noexcept
   {
-    static_assert(std::is_constructible<T, Args&&...>::value, "T must be constructible with Args");
+    static_assert(sizeof...(Args) == 1, "Args must be a single lvalue");
+    static_assert(std::is_lvalue_reference<Args...>::value, "Args must be a single lvalue");
 
-    *this = nullopt;
-    this->construct(std::forward<Args>(args)...);
+    m_value = thrust::addressof(args...);
+    return **this;
```

Comment:
I would simplify this to
```suggestion
    return *m_value;
```